File: entrega2/menu/src/main/java/com/aps/menu/cadastros/CadastroRestaurante.java
package com.aps.menu.cadastros;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.aps.menu.entidades.Conta;
import com.aps.menu.entidades.Restaurante;

@Component
public class CadastroRestaurante {
    @Autowired
    private IRepositorioRestaurante repositorioRestaurante;

    public boolean existeRestaurante(Conta conta) {
        return repositorioRestaurante.existe(conta);
    }

    public Restaurante registrar(Restaurante restaurante) {
        return repositorioRestaurante.registrar(restaurante);
    }
}

File: rup/menu-app/src/cadastros/cadastroRestaurante.ts
import { Conta } from "../entidades/conta";
import { Restaurante } from "../entidades/restaurante";
import { IRepositorioRestaurantes } from "./iRepositorioRestaurantes";

export class CadastroRestaurante {
    private repositorio: IRepositorioRestaurantes;

    constructor(repositorio: IRepositorioRestaurantes) {
        this.repositorio = repositorio;
    }

    existeRestaurante(c: Conta): boolean {
        return this.repositorio.existeRestaurante(c);
    }

    registrarRestaurante(r: Restaurante): Restaurante {
        return this.repositorio.registrarRestaurante(r);
    }
}